signac/core/collection_api.py

```diff
@@ -0,0 +1,446 @@
+import os
+import json
+import errno
+import uuid
+from copy import copy
+from contextlib import contextmanager
+from collections.abc import Mapping
+from collections.abc import MutableMapping
+from collections.abc import Sequence
+from collections.abc import MutableSequence
+
+try:
+    import numpy
+    NUMPY = True
+except ImportError:
+    NUMPY = False
+
+try:
+    from collections.abc import Collection
+except ImportError:
+    # create Collection
```

Comment:
What’s the case that `Collection` can’t be imported? Is it only present in specific Python versions?

Reply:
`Collection` is not present in `Python3.5`. They added it 3.6 so need to define it for Pyhton3.5

Reply:
This is a good solution! Python 3.5 will be end-of-life in September, so we may refactor and remove support for Python 3.5 someday. Please add a comment (suggestion below) indicating why you created the fallback. It's important that the comment includes the words "3.5" and "3.6" so we know when this support can be removed.
```suggestion
    # Collection does not exist in Python 3.5, only Python 3.6 or newer.
```

Reply:
Sorry if this caused confusion, I suggested this to Vishav a while ago.